### DISCUSSION TITLE
useCombobox vs useSearchAutocomplete

### DISCUSSION THREAD

**Discussion by jenmak-tangelo** [adobe/react-spectrum#4724] [Q&A]
Hi,
I am wondering when it is appropriate to use the useComboBox and when to use the useSearchAutocomplete. What is the main difference between the two?

Thank you!

**Comment by reidbarber** [comment]
The main difference is that useComboBox has a button attached to it that will open the popover. Also useSearchAutocomplete includes a clear button to clear the field.

I would say that useComboBox is more like a select/picker component that supports filtering down options (although it does also support custom values), where useSearchAutocomplete is more for showing suggestions in a search input field. Both use the combobox ARIA pattern. You'll notice that under the hood, useSearchAutocomplete uses both useComboBox and useSearchField.
